torch/lib/c10d/ProcessGroup.cpp

```diff
@@ -51,10 +53,24 @@ bool isP2POp(OpType opType) {
       opType == OpType::RECVANYSOURCE;
 }
 
-ProcessGroup::Work::Work() : rank_(-1), opType_(OpType::UNKNOWN) {}
 
-ProcessGroup::Work::Work(int rank, OpType opType)
-    : rank_(rank), opType_(opType) {}
+ProcessGroup::Work::Work(int rank, OpType opType, const char* profiling_title)
+    : rank_(rank), opType_(opType) {
+  if (profiling_title != nullptr) {
+    profilingFuture_ = c10::make_intrusive<c10::ivalue::Future>(
+        c10::TensorType::get());
+    auto recordingFunction = std::make_shared<at::RecordFunction>(at::RecordScope::USER_SCOPE);
+    if (recordingFunction->active) {
+        recordingFunction->before(profiling_title, {});
+        std::function<void()> end_handler = [this, recordingFunction]() {
```

Comment:
Can we `std::move(recordingFunction)` since it's not used after this anymore to avoid a copy?

Reply:
It is shared_ptr and copy is cheap. right after this block also the extra copy is destroyed. If I wanted to use std:move line 62 will change to something much less readable:
` std::function<void()> end_handler = [this, recordingFunction{std::move(recordingFunction)}]()`